python/cudf/cudf/tests/test_parquet.py

```diff
@@ -437,3 +471,26 @@ def test_multifile_warning(datadir):
         expect = pd.read_parquet(fname)
         expect = expect.apply(pd.to_numeric)
         assert_eq(expect, got)
+
+
+# Validates the integrity of the GPU accelerated parquet writer.
+def test_parquet_writer_gpu(tmpdir, simple_gdf):
+    gdf_fname = tmpdir.join("gdf.parquet")
+
+    # Write out the gdf using the GPU accelerated writer
+    simple_gdf.to_parquet(gdf_fname.strpath, engine="cudf")
+    assert os.path.exists(gdf_fname)
+    expect = simple_gdf
+
+    # Read the gdf back in using the reader
+    got = cudf.read_parquet(gdf_fname)
+
+    # There is a special case in the reader that causes the 'test_index'
+    # Index used by Pandas to be read in as a normal data column.
+    # Therefore it must be removed here and the index name set to
+    # that to match the "expected" pdf
+    if "test_index" in got.columns:
+        got = got.drop("test_index")
+    got.index.name = "test_index"
```

Comment:
I'm confused by this logic being in the test. Is this something we should handle in the writer to not end up with an erroneous `test_index` column?

Reply:
@kkraus14 I am going to try this approach mentioned by @OlivierNV in the reader, not writer. This logic was added in the case due to the "test_index" column being considered a data column by the reader. Therefore without dropping the column in the test the test would fail. Should have another commit here shortly and hopefully this does the trick ... fingers crossed!

> If adding the index column, in order for the index not to appear as a data column, we might also need to populate the metadata 'pandas' string (see get_pandas_index_name() in the io reader)
Something like "pandas" = "{"index_columns": ["test_index"]}" (Although, looking at the code, it seems that the index is only excluded if loading specific column names). It seems that extracting the index should be done on the python side, though.